daisy/workflow.go

```diff
@@ -247,7 +257,23 @@ func (w *Workflow) RunWithModifiers(
 	return nil
 }
 
+func (w *Workflow) recordStepTime(stepName string, startTime time.Time, endTime time.Time) {
+	if w.parent == nil {
+		w.recordTimeMx.Lock()
+		w.stepTimeRecords = append(w.stepTimeRecords, TimeRecord{stepName, startTime, endTime})
+		w.recordTimeMx.Unlock()
+	} else {
+		w.parent.recordStepTime(fmt.Sprintf("%s.%s", w.Name, stepName), startTime, endTime)
```

Comment:
Can we  have nested workflows with more than one level of nesting? E.g. w.parent.parent != nil. If so, should we recurse  up the graph until we find the root instead of just doing one step up?

Reply:
This is a recursive call: w.parent.recordStepTime